tests/trainer/test_trainer.py

```diff
@@ -460,6 +510,54 @@ def on_load_checkpoint(self, _):
         assert next_model.num_on_load_checkpoint_called == 1
 
 
+def test_trainer_properties_resume_from_checkpoint(tmpdir):
+    """Test that required trainer properties are set correctly when resuming from checkpoint in different
+    phases."""
+
+    class CustomBoringModel(BoringModel):
+        def __init__(self, lr=1e-2):
+            super().__init__()
+            self.save_hyperparameters()
+
+        def configure_optimizers(self):
+            optimizer = torch.optim.SGD(self.layer.parameters(), lr=self.hparams.lr)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+            return [optimizer], [lr_scheduler]
+
+    model = CustomBoringModel()
+    checkpoint_callback = ModelCheckpoint(dirpath=tmpdir, monitor="val_loss", save_last=True)
+    trainer_args = dict(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=7,
+        limit_val_batches=7,
+        logger=False,
+        callbacks=[checkpoint_callback],  # this performs the assertions
```

Comment:
the `ModelCheckpoint` performs the assertions?

Reply:
Copy paste 😅.. will update